pennylane/beta/interfaces/tf.py

```diff
@@ -0,0 +1,187 @@
+
+    .. code-block:: python
+
+        class MyAutogradQuantumTape(AutogradInterface, QuantumTape):
+
+    Alternatively, the autograd interface can be dynamically applied to existing
+    quantum tapes via the :meth:`~.apply` class method. This modifies the
+    tape **in place**.
+
+    Once created, the autograd interface can be used to perform quantum-classical
+    differentiable programming.
+
+    .. note::
+
+        If using a device that supports native autograd computation and backpropagation, such as
+        :class:`~.DefaultQubitTF`, the TensorFlow interface **does not need to be applied**. It is
+        only applied to tapes executed on non-TensorFlow compatible devices.
+
+    **Example**
+
+    One a TensorFlow quantum tape has been created, it can be differentiated using the gradient tape:
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit", wires=1)
+        p = tf.Variable([0.1, 0.2, 0.3], dtype=tf.float64)
+
+        with tf.GradientTape() as tape:
+            with TFInterface.apply(QuantumTape()) as qtape:
+                qml.Rot(p[0], p[1] ** 2 + p[0] * p[2], p[1] * tf.sin(p[2]), wires=0)
+                expval(qml.PauliX(0))
+
+            result = qtape.execute(dev)
+
+    >>> print(result)
+    tf.Tensor([0.06982072], shape=(1,), dtype=float64)
+    >>> grad = tape.gradient(result, p)
+    >>> print(grad)
+    tf.Tensor([0.29874274 0.39710271 0.09958091], shape=(3,), dtype=float64)
+
+    The TensorFlow interface defaults to ``tf.float64`` output. This can be modified by
+    providing the ``dtype`` argument when applying the interface:
+
+    >>> p = tf.Variable([0.1, 0.2, 0.3], dtype=tf.float32)
+    >>> with tf.GradientTape() as tape:
+    ...     TFInterface.apply(qtape, dtype=tf.float32)  # reusing the previous qtape
+    ...     result = qtape.execute(dev)
+    >>> print(result)
+    tf.Tensor([0.06982072], shape=(1,), dtype=float32)
+    >>> grad = tape.gradient(result, p)
+    >>> print(grad)
+    tf.Tensor([0.2895088  0.38464668 0.09645163], shape=(3,), dtype=float32)
+    """
+
+    dtype = tf.float64
+
+    @property
+    def interface(self):  # pylint: disable=missing-function-docstring
+        return "tf"
+
+    def _update_trainable_params(self):
+        params = self.get_parameters(trainable_only=False)
+
+        trainable_params = set()
+
+        for idx, p in enumerate(params):
+            # Determine which input tensors/Variables are being recorded for backpropagation.
+            # The function should_record_backprop, documented here:
+            # https://github.com/tensorflow/tensorflow/tree/master/tensorflow/python/eager/tape.py#L163
+            # accepts lists of *tensors* (not Variables), returning True if all are being watched by one or more
+            # existing gradient tape, False if not.
+            if isinstance(p, (tf.Variable, tf.Tensor)) and should_record_backprop(
+                [tf.convert_to_tensor(p)]

```

Comment:
is this conversion necessary?

Reply:
Yep. `should_record_backprop` for some reason won't accept `tf.Variable` objects, just `tf.Tensor` objects. It's very annoying.

Reply:
Especially because it's not documented, so it took a bit of trial and error to work out why it was always returning `False`! 😆 

I don't think this should have too much overhead though --- converting from a `tf.Variable` to a `tf.Tensor` should result in no modification/copying of the underlying data.